awswrangler/_utils.py

```diff
@@ -414,7 +415,7 @@ def table_refs_to_df(
 ) -> pd.DataFrame:
     """Build Pandas dataframe from list of PyArrow tables."""
     if isinstance(tables[0], pa.Table):
-        return ensure_df_is_mutable(pa.concat_tables(tables, promote=True).to_pandas(**kwargs))
```

Comment:
This is a change that I would like to discuss. The column manipulations in the [`ensure_df_is_mutable`](https://github.com/awslabs/aws-data-wrangler/blob/main/awswrangler/_utils.py#L308) would be too slow on a large DataFrame in the `distributed` case, so I had to remove it

Reply:
+1 we probably shouldn't be doing that in distributed scenario